apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,61 +1,55 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+  Component,
+  ContentChild,
+  Directive,
+  EventEmitter,
+  Input,
+  Output,
+  TemplateRef,
+} from '@angular/core';
 import { ListItemComponent } from '../list-item/list-item.component';
 
+@Directive({
+  selector: 'app-card-image',
+  standalone: true,
+})
+export class CardImageDirective {}
+
+@Directive({
+  selector: '[appCardListItem]',
+  standalone: true,
+})
+export class CardListItemDirective<T> {
+  constructor(public templateRef: TemplateRef<{ $implicit: T }>) {}
+}
+
 @Component({
   selector: 'app-card',
   template: `
-    <div
-      class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
-      [class]="customClass">
-      <img
-        *ngIf="type === CardType.TEACHER"
-        src="assets/img/teacher.png"
-        width="200px" />
-      <img
-        *ngIf="type === CardType.STUDENT"
-        src="assets/img/student.webp"
-        width="200px" />
+    <div class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4">
+      <ng-content select="app-card-image"></ng-content>
 
       <section>
-        <app-list-item
-          *ngFor="let item of list"
-          [name]="item.firstName"
-          [id]="item.id"
-          [type]="type"></app-list-item>
+        @for (item of list; track $index) {
```

Comment:
be careful with $index. If the list changed, the $index is not efficient.
you should always try to track a property of your list and use $index for list that doesn't change

Reply:
I really appreciate your observation. At this case, generic T has no explicit properties, it is correct if track with whole item?:  @for (item of list; track item)

Reply:
You can do <T extends {id: number)> so each element has to have an id.